lib/elastic_apm/metrics/vm.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module ElasticAPM
+  module Metrics
+    # @api private
+    class VM
+      def initialize(_config)
+        @total_time = 0
+      end
+
+      # rubocop:disable Metrics/MethodLength
+      def collect
+        stat = GC.stat
+        @total_time += GC::Profiler.total_time
+        GC::Profiler.clear
+        thread_count = Thread.list.count
+
+        {
+          'ruby.gc.count': stat[:count],
+          'ruby.gc.time': @total_time,
+          'ruby.heap.live': stat[:heap_live_slots],
```

Comment:
Repeating my message from Slack, for posterity: let's call this `ruby.heap.slots.live`, and `ruby.heap.slots.free` below.